src/main/java/net/neoforged/neoforge/capabilities/RegisterCapabilitiesEvent.java

```diff
@@ -79,6 +80,34 @@ public boolean isBlockRegistered(BlockCapability<?, ?> capability, Block block)
         return capability.providers.containsKey(block);
     }
 
+    /**
+     * Makes a block capability proxyable.
+     *
+     * <p>This method should only be called by the mod that defines the capability,
+     * in the {@link EventPriority#HIGH} or {@link EventPriority#HIGHEST} phase.
+     *
+     * @throws IllegalStateException if the capability was marked to be non-proxyable
+     * @see BlockCapability#isProxyable()
+     */
+    public void setProxyable(BlockCapability<?, ?> capability) {
+        capability.setProxyable(true);
+    }
+
+    /**
+     * Makes a block capability non-proxyable.
+     * This should typically be called when unexpectedly proxying the capability (e.g. to a different block position)
+     * would cause crashes, for example because the returned capability is tied to the query position.
+     *
+     * <p>This method should only be called by the mod that defines the capability,
+     * in the {@link EventPriority#HIGH} or {@link EventPriority#HIGHEST} phase.
+     *
+     * @throws IllegalStateException if the capability was marked to be proxyable
```

Comment:
I think this would be clearer as `@throws IllegalStateException if the capability was already marked as proxyable`